extensions/interactions/CodeRepl/CodeRepl.js

```diff
@@ -136,6 +136,22 @@ oppia.directive('oppiaInteractiveCodeRepl', [
           });
         };
 
+        $scope.testRun = function(codeInput) {
```

Comment:
This looks like it'll work fine, but it duplicates most of runCode(). This is generally not a good practice when coding, since it causes skew to happen (i.e., if a developer edits one of the two pieces of code, and forgets to edit the other, that's an easy way for an error to occur).

To prevent against this, we try to follow the DRY (don't repeat yourself) principle. So, see if you can find a way to refactor this so that we aren't repeating code.

You can start by renaming the previous function to $scope.runAndSubmitCode(), and naming this function $scope.runCode() (since that's what each is doing). Then find a way to have $scope.runAndSubmitCode() make a call to $scope.runCode() to do most of the work, and pass in an onFinishEval callback that sends the response.